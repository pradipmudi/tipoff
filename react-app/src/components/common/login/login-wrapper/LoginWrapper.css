.login-box{
    background-color: #e1f4f3 !important;
}

.login-btn{
    background-color: #333333 !important;
}


