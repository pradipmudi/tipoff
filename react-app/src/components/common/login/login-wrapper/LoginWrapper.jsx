import React, { useState } from 'react';
import {
    Hero,
    HeroBody,
    Container,
    Columns,
    Column,
    TabList,
    TabLink,
    Tabs,
    Tab,
    HeroHeader
} from 'bloomer';

import LoginForm from '../login-form/LoginForm';
import RegistrationForm from '../registration-form/RegistrationForm';
import {TabContentRenderer} from '../../tab-content-renderer/TabContentRenderer';

import './LoginWrapper.css';

const LoginWrapper = () => {

    const [state,setState] = useState({
        tab:"login",
        active:"login",
        tabs :{
            login : LoginForm,
            registration: RegistrationForm
        }
    });

    const handleTab = evt => {
        setState({
            ...state,
            active: evt.currentTarget.dataset.id,
            tab: evt.currentTarget.dataset.id
        });
    };
    
    return (
        <Hero isFullHeight >
            <HeroHeader>
            </HeroHeader>
            <HeroBody >
                <Container hasTextAlign='centered'>
                <Columns isCentered>
                    <Column isSize="1/2">
                        <Tabs >
                            
                                <TabList >
                                    <Tab  isActive={state.active === "login"} 
                                        data-id="login"
                                        onClick={e => handleTab(e)}
                                    >
                                        <TabLink>
                                            <span>Login</span>
                                        </TabLink>
                                    </Tab>
                                    <Tab isActive={state.active === "registration"}
                                        data-id="registration"
                                        onClick={e => handleTab(e)}
                                    >
                                        <TabLink>
                                            <span>Registration</span>
                                        </TabLink>
                                    </Tab> 
                                </TabList>
                                <hr/>
                            
                        
                        </Tabs>
                        
                        <TabContentRenderer tabs={state.tabs} tab={state.tab} />
                    </Column>
                </Columns>
                    
                </Container>
            </HeroBody>
        </Hero>
    )
    
}

export default LoginWrapper;
