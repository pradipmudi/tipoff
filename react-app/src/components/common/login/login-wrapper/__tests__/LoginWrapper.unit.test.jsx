import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import {Provider} from 'react-redux';
import store from '../../../../../redux/store';

import LoginWrapper from '../LoginWrapper';

describe("LoginWrapper", () => {
    it("matches snapshot", () => {
        const snap = renderer.create(
            <Provider store={store}>
                <MemoryRouter>
                    <LoginWrapper />
                </MemoryRouter>
            </Provider>
        ).toJSON();
        expect(snap).toMatchSnapshot();
    });

    it("renders without issues", () => {
        const wrapper = shallow(<LoginWrapper />);
        expect(wrapper.exists()).toEqual(true);
    });
});