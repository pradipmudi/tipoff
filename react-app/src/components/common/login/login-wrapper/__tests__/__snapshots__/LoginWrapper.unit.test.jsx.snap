// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginWrapper matches snapshot 1`] = `
<section
  className="hero is-fullheight"
>
  <header
    className="hero-head"
  />
  <div
    className="hero-body"
  >
    <div
      className="container has-text-centered"
    >
      <div
        className="columns is-centered"
      >
        <div
          className="column is-half"
        >
          <div
            className="tabs"
          >
            <ul>
              <li
                className="is-active"
                data-id="login"
                onClick={[Function]}
              >
                <a>
                  <span>
                    Login
                  </span>
                </a>
              </li>
              <li
                data-id="registration"
                onClick={[Function]}
              >
                <a>
                  <span>
                    Registration
                  </span>
                </a>
              </li>
            </ul>
            <hr />
          </div>
          <div>
            <div
              className="box login-box"
            >
              <div>
                <div
                  className="notification is-danger is-hidden tipoff-notification"
                >
                  <span />
                </div>
              </div>
              <div
                className="field is-horizontal"
              >
                <div
                  className="field-body"
                >
                  <div
                    className="field"
                  >
                    <div
                      className="control"
                    >
                      <input
                        className="input"
                        name="userId"
                        onChange={[Function]}
                        placeholder="Email"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="field is-horizontal"
              >
                <div
                  className="field-body"
                >
                  <div
                    className="field"
                  >
                    <div
                      className="control"
                    >
                      <input
                        className="input"
                        name="password"
                        onChange={[Function]}
                        placeholder="Password"
                        type="password"
                        value=""
                      />
                    </div>
                  </div>
                </div>
              </div>
              <button
                className="button is-large is-fullwidth login-btn"
                onClick={[Function]}
                type="button"
              >
                <label
                  className="label has-text-white"
                >
                  Login 
                  <span
                    className="icon"
                    style={
                      Object {
                        "marginLeft": ".1em",
                      }
                    }
                  >
                    <i
                      aria-hidden="true"
                      className="fas fa-sign-in-alt"
                    />
                  </span>
                </label>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
